src/app/app.component.ts

```diff
@@ -261,14 +341,79 @@ export class AppComponent implements AfterViewInit {
     this.entrySelected = treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
   }
 
+  async onToolbarOptionSelected(event: CustomEvent) {
+    await event.detail.tag.handler();
+  }
+
+  openNewFolderDialog(): void {
+    this.dialog.open(NewFolderModalComponent, {
+      data: {makeNewFolder: this.makeNewFolder.bind(this) },
+    });
+  }
+
+  openEditColumnsDialog(): void {
+    this.dialog.open(EditColumnsModalComponent, {
+      data: { columnsSelected: this.selectedColumns, allColumnOptions: this.allPossibleColumns, updateColumns: this.updateColumns.bind(this),},
+    });
+  }
+
+  async makeNewFolder(folderName: string) {
+    if (folderName) {
+      if (!this.lfRepositoryBrowser?.nativeElement?.currentFolder) {
+        throw new Error(
+          this.localizationService.getString('NO_CURRENTLY_OPENED_FOLDER')
+        );
+      }
+      await this.addNewFolderAsync(this.lfRepositoryBrowser?.nativeElement?.currentFolder, folderName);
+      await this.lfRepositoryBrowser?.nativeElement?.refreshAsync();
+    }
+    else {
+      throw new Error(
+        this.localizationService.getString('PLEASE_PROVIDE_FOLDER_NAME')
+      );
+    }
+
+  }
+
+  async addNewFolderAsync(parentNode: LfTreeNode, folderName: string): Promise<void> {
+    if (!this.repoClient){
+      throw new Error('repoClient is undefined');
+    }
+    type RequestParameters = { entryId: number; postEntryChildrenRequest: PostEntryChildrenRequest};
+    const entryId = (parentNode as LfRepoTreeNode).targetId ?? parseInt( parentNode.id, 10);
+    const postEntryChildrenRequest: PostEntryChildrenRequest = new PostEntryChildrenRequest({
+      name: folderName,
+      entryType: PostEntryChildrenEntryType.Folder,
+    });
+    const requestParameters: RequestParameters = {
+      entryId,
+      postEntryChildrenRequest,
+    };
+    const repoId: string = await this.repoClient.getCurrentRepoId();
+    await this.repoClient?.entriesClient.createOrCopyEntry(
+      {
+        repoId,
+        entryId: requestParameters.entryId,
```

Comment:
unless I'm missing something you don't need to create a typed object for this, you can just assign the variable directly